### DISCUSSION TITLE
Unbounded variable `status_code` openapi utils.py

### DISCUSSION THREAD

**Discussion by jimixxperez** [fastapi/fastapi#10639] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from collections.abc import AsyncIterable
import json
from typing import Generic, TypeVar, Mapping
from starlette.background import BackgroundTask
from starlette.responses import StreamingResponse
from fastapi.encoders import jsonable_encoder

from pydantic import BaseModel

NDJSONPayload = TypeVar("NDJSONPayload", bound=BaseModel)


class StreamingNDJSONResponse(Generic[NDJSONPayload], StreamingResponse):
    """Converts a pydantic model generator into a streaming HTTP Response
    that streams NDJSON.

    """

    def __init__(
        self,
        content_generator: AsyncIterable[NDJSONPayload],
        status_code: int = 200,
        headers: Mapping[str, str] | None = None,
        background: BackgroundTask | None = None,
    ) -> None:
        super(StreamingResponse, self).__init__(
            content=self._encoded_generator(content_generator),
            status_code=status_code,
            headers=headers,
            media_type="application/x-ndjson",
            background=background,
        )

    async def _encoded_generator(self, content_generator: AsyncIterable[NDJSONPayload]):
        async for item in content_generator:
            yield json.dumps(jsonable_encoder(item)) + "\n"

from fastapi import FastAPI

app = FastAPI()


@app.get("/", response_class=StreamingNDJSONResponse[str])
async def root() -> StreamingNDJSONResponse[str]:
    ...
```

### Description

While I was working on custom StreamingReponse type. It seems as if the missing `status_code` in the aforementioned type  triggered the unbounded variable issue during the inspection of the response class.
By looking at the source code it seems as if the creation and assignment of the variable is conditionally bounded.
https://github.com/tiangolo/fastapi/blob/480620372a662aa9025c47410fbc90a255b2fc94/fastapi/openapi/utils.py#L313

### Operating System

Linux

### Operating System Details

Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

3.10.12

**Comment by YuriiMotov** [comment, marked as answer]
The problem is that FastAPI doesn't recognize that `StreamingNDJSONResponse[str]` is subclass of `Response` and treats it as response model.
Not sure why it happens.

But you can solve this by either removing return type annotation:
```
async def root():
```
or changing it to base class:
```
async def root() -> StreamingResponse:
```
or specifying response model:
```
@app.get("/", response_class=StreamingNDJSONResponse[str], response_model=None)
```

Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
